src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -3020,19 +3044,31 @@ ReceiveResults(WorkerSession *session, bool storeRows)
 	uint32 expectedColumnCount = 0;
 	char **columnArray = NULL;
 	Tuplestorestate *tupleStore = execution->tupleStore;
+	MemoryContext ioContext = NULL;
+
+	/*
+	 * These fields are computed once at the start of the execution for performance
+	 * reasons. So, do not (p)free these fields, these are set to local variables for
+	 * readability purpose.
+	 */
+	attributeInputMetadata = execution->attributeInputMetadata;
+	columnArray = execution->columnArray;
```

Comment:
This could probably move to the initialisation section. I'm not sure the comments will add much. It's kind of obvious that you should not pfree things from the execution.